File: Array/leetcode_3238.cpp

// Problem link ===>> https://leetcode.com/problems/find-the-number-of-winning-players/description/

#include <bits/stdc++.h>
#define ll long long
using namespace std;

class Solution
{
public:
    int winningPlayerCount(int n, vector<vector<int>> &pick)
    {
        int ans = 0;
        for (int i = 0; i < n; i++){
            map<int, int> mp;
            for (int j = 0; j < pick.size(); j++)
            {
                if(pick[j][0]==i){
                    mp[pick[j][1]]++;
                    if (mp[pick[j][1]]>i){
                        ans++;
                        break;
                    }
                }
            }
        }
        return ans;
    }
};

int main()
{
    ll t;
    cin >> t;
    while (t--)
    {
        ll n, m;
        cin >> n >> m;
        vector<vector<int>> pick(m, vector<int>(2));
        for (int i = 0; i < m; i++)
        {
            cin >> pick[i][0] >> pick[i][1];
        }
        Solution sol;
        cout << sol.winningPlayerCount(n, pick) << endl;
    }
    return 0;
}

File: Array/leetcode_3238.py

# Problem link ===>>  https://leetcode.com/problems/find-the-number-of-winning-players/description/


class List:
    def __init__(self):
        self.data = []

    def append(self, item):
        self.data.append(item)


class Solution:
    def winningPlayerCount(self, n: int, pick: List[List[int]]) -> int:
        ans = 0
        for i in range(n):
            mp = {}
            for j in range(len(pick)):
                if pick[j][0] == i:
                    mp[pick[j][1]] = mp.get(pick[j][1], 0) + 1
                    if mp[pick[j][1]] > i:
                        ans += 1
                        break
        return ans

def main():
    t = int(input())
    for _ in range(t):
        n, m = map(int, input().split())
        pick = []
        for _ in range(m):
            pick.append(list(map(int, input().split())))
        sol = Solution()
        print(sol.winningPlayerCount(n, pick))


if __name__ == "__main__":
    solution = Solution()
